### PR TITLE
Add internal link

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: BackofficeBundle/Resources/public/coffee/InternalLinkFormView.coffee [hunk 1]
```diff
@@ -0,0 +1,58 @@
+InternalLinkFormView = OrchestraView.extend(
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/InternalLinkFormView.coffee]: can use namespace, on the syntax with class for this backbone view and add js comment for methods view

**Comment by itkg-nanne** [comment]
done @itkg-alavieille

**Comment by itkg-alavieille** [comment]
:+1:  for this part

File: Backoffice/EventSubscriber/SiteSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * @param FormEvent $event
+     */
+    public function preSubmit(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $event->getData();
+        if (!is_null($data) && $data['siteId'] != '') {
+            $form->add('siteAlias', 'choice', array(
+                    'label' => 'open_orchestra_backoffice.form.internal_link.site_alias',
```
Comment (nicolasThal) [inline Backoffice/EventSubscriber/SiteSubscriber.php]: could you correct the indentation ?

File: Backoffice/Tests/EventSubscriber/ContentSearchSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    /**
+     * Test preSubmit
+     */
+    public function testPreSubmit()
+    {
+        $this->subscriber->preSubmit($this->event);
+
+        Phake::verify($this->form)->add('contentId', 'choice', array(
+                    'label' => false,
```
Comment (nicolasThal) [inline Backoffice/Tests/EventSubscriber/ContentSearchSubscriberTest.php]: could you fix the indentation ?

**Comment by itkg-nanne** [comment]
done @nicolasThal

File: Backoffice/EventSubscriber/SiteSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        $choices = array();
+        $site = $this->siteRepository->findOneBySiteId($siteId);
+        $aliases = $site->getAliases();
+        foreach ($aliases as $alias) {
+            $choices[$alias->getDomain()] = $alias->getDomain();
+        }
+
+        return $choices;
+    }
+
```
Comment (nicolasThal) [inline Backoffice/EventSubscriber/SiteSubscriber.php]: empty line to suppress

File: Backoffice/EventSubscriber/ContentSearchSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        $form = $event->getForm();
+        $data = $event->getData();
+        if (!is_null($data) && ($data['contentType'] != '' || $data['keywords'] != '')) {
+            $condition = null;
+            if ($data['keywords'] != '') {
+                $condition = json_decode($this->transformer->reverseTransform($data['keywords']), true);
+            }
+            $form->add('contentId', 'choice', array(
+                    'label' => false,
```
Comment (nicolasThal) [inline Backoffice/EventSubscriber/ContentSearchSubscriber.php]: i missed this indentation to fix ..

File: Backoffice/Form/Type/Component/InternalLinkType.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    /**
+     * @return string The name of this type
+     */
+    public function getName()
+    {
+        return "oo_internal_link";
+    }
+
```
Comment (nicolasThal) [inline Backoffice/Form/Type/Component/InternalLinkType.php]: empty line to suppress

File: Backoffice/EventSubscriber/SiteSubscriber.php (hunk 4 above)
Comment (itkg-nanne) [inline Backoffice/EventSubscriber/SiteSubscriber.php]: done @nicolasThal

**Comment by nicolasThal** [comment]
:+1: with the little checkstyles

**Pull request merged**